DrawerKit/DrawerKit/Internal API/PresentationController.swift

```diff
@@ -167,31 +286,30 @@ private extension PresentationController {
 }
 
 private extension PresentationController {
-    func animateTransition(to endPositionY: CGFloat, clamping: Bool = false) {
-        guard endPositionY != currentDrawerY else { return }
-
-        let endPosY = (clamping ? clamped(endPositionY) : endPositionY)
-        guard endPosY != currentDrawerY else { return }
+    func animateTransition(to endState: DrawerState) {
+        guard endState != currentDrawerState else { return }
 
         let animator = UIViewPropertyAnimator(duration: durationInSeconds,
                                               timingParameters: timingCurveProvider)
 
         let maxCornerRadius = maximumCornerRadius
-        let endingCornerRadius = cornerRadius(at: endPosY)
+        let endingCornerRadius = cornerRadius(at: endState)
+        let endingPositionY = drawerPositionY(for: endState)
+
         animator.addAnimations {
-            self.currentDrawerY = endPosY
-            if maxCornerRadius > 0 {
+            self.currentDrawerY = endingPositionY
+            if maxCornerRadius != 0 {
                 self.currentDrawerCornerRadius = endingCornerRadius
             }
         }
 
-        if endPosY == containerViewH {
+        if endState == .collapsed {
             animator.addCompletion { _ in
                 self.presentedViewController.dismiss(animated: true)
             }
         }
 
-        if maxCornerRadius > 0 && endPosY != drawerPartialY {
+        if maxCornerRadius != 0 && (endState == .collapsed || endState == .fullyExpanded) {
```

Comment:
here I like them :D

Reply:
Here they are needed, else the boolean logic would have a different result.